packages-exp/app-compat/src/firebaseNamespaceCore.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  _FirebaseApp,
+  _FirebaseNamespace,
+  FirebaseService,
+  FirebaseServiceNamespace
+} from '@firebase/app-types/private';
+import {
+  SDK_VERSION,
+  initializeApp,
+  registerVersion,
+  _registerComponent,
+  _DEFAULT_ENTRY_NAME
+} from '@firebase/app-exp';
+import { _FirebaseAppInternal } from '@firebase/app-types-exp';
+import { Component, ComponentType } from '@firebase/component';
+
+import { deepExtend, contains } from '@firebase/util';
+import { FirebaseAppImpl } from './firebaseApp';
+import { ERROR_FACTORY, AppError } from './errors';
+import { FirebaseAppLiteImpl } from './lite/firebaseAppLite';
+
+/**
+ * Because auth can't share code with other components, we attach the utility functions
+ * in an internal namespace to share code.
+ * This function return a firebase namespace object without
+ * any utility functions, so it can be shared between the regular firebaseNamespace and
+ * the lite version.
+ */
+export function createFirebaseNamespaceCore(
+  firebaseAppImpl: typeof FirebaseAppImpl | typeof FirebaseAppLiteImpl
+): FirebaseNamespace {
+  const apps: { [name: string]: FirebaseApp } = {};
+  // // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  // const components = new Map<string, Component<any>>();
+
+  // A namespace is a plain JavaScript Object.
+  const namespace: FirebaseNamespace = {
+    // Hack to prevent Babel from modifying the object returned
+    // as the firebase namespace.
+    // @ts-ignore
+    __esModule: true,
+    initializeApp: initializeAppCompat,
+    // @ts-ignore
+    app,
+    registerVersion,
+    // @ts-ignore
+    apps: null,
+    SDK_VERSION,
+    INTERNAL: {
+      registerComponent: registerComponentCompat,
+      removeApp,
+      useAsService
+    }
+  };
+
+  // Inject a circular default export to allow Babel users who were previously
+  // using:
+  //
+  //   import firebase from 'firebase';
+  //   which becomes: var firebase = require('firebase').default;
+  //
+  // instead of
+  //
+  //   import * as firebase from 'firebase';
+  //   which becomes: var firebase = require('firebase');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (namespace as any)['default'] = namespace;
+
+  // firebase.apps is a read-only getter.
+  Object.defineProperty(namespace, 'apps', {
+    get: getApps
+  });
+
+  /**
+   * Called by App.delete() - but before any services associated with the App
+   * are deleted.
+   */
+  function removeApp(name: string): void {
+    delete apps[name];
+  }
+
+  /**
+   * Get the App object for a given name (or DEFAULT).
+   */
+  function app(name?: string): FirebaseApp {
+    name = name || _DEFAULT_ENTRY_NAME;
+    if (!contains(apps, name)) {
+      throw ERROR_FACTORY.create(AppError.NO_APP, { appName: name });
+    }
+    return apps[name];
+  }
+
+  // @ts-ignore
+  app['App'] = firebaseAppImpl;
+
+  /**
+   * Create a new App instance (name must be unique).
+   */
+  function initializeAppCompat(
```

Comment:
Doesn't accept a string as a second arg anymore?

Reply:
It does. `FirebaseNamespace` defines the signature for `firebase.initializeApp` which I didn't change.